pkg/codegen/testing/test/testdata/regress-go-12971/schema.json

```diff
@@ -0,0 +1,20 @@
+{
+  "name": "world",
+  "version": "0.0.1",
+  "types": {
+    "world::World": {
+      "type": "object",
+      "properties": {
+        "name": {"type": "string", "defaultInfo": {"environment": ["WORLD_NAME"]}},
+        "radiusKm": {"type": "number", "defaultInfo": {"environment": ["WORLD_RADIUS_KM"]}},
+        "populated": {"type": "boolean", "defaultInfo": {"environment": ["WORLD_POPULATED"]}}
```

Comment:
Should we also include a `"provider"` resource with similar input properties to these, to ensure the generated  `NewProvider` func is what we expect (which is where the problem is in this case for the Snowflake provider)? Right now, we're only testing the generated `Default` method for types.

Something like this in the schema:

```json
"types": {
    ...
},
"provider": {
    "inputProperties": {
        "name": {"type": "string", "defaultInfo": {"environment": ["WORLD_NAME"]}},
        "radiusKm": {"type": "number", "defaultInfo": {"environment": ["WORLD_RADIUS_KM"]}},
        "populated": {"type": "boolean", "defaultInfo": {"environment": ["WORLD_POPULATED"]}}
    }
}
```

Reference: https://github.com/pulumi/pulumi-snowflake/blob/0742790b87a393e48b7b2483c7cb0837cd3bc4b5/provider/cmd/pulumi-resource-snowflake/schema.json#L1993-L2002

Reply:
Good call. I'm changing the test case to verify provider as well as config.
That should cover all cases.